api/scpca_portal/models/project.py

```diff
@@ -1155,6 +1101,171 @@ def update_spatial_data(future):
+            sample_id = sample_metadata["scpca_sample_id"]
+
+            # Some samples will exist but their contents cannot be shared yet.
+            # When this happens their corresponding sample folder will not exist.
+            sample_dir = self.get_sample_input_data_dir(sample_id)
+
+            has_bulk_rna_seq = sample_id in bulk_rna_seq_sample_ids
+            has_cite_seq_data = len(list(Path(sample_dir).glob("*_adt.*"))) > 0
+            has_single_cell_data = len(list(Path(sample_dir).glob("*_metadata.json"))) > 0
+            has_spatial_data = len(list(Path(sample_dir).rglob("*_spatial/*_metadata.json"))) > 0
+            include_anndata = len(list(Path(sample_dir).glob("*.hdf5"))) > 0
+
+            sample_metadata["age_at_diagnosis"] = sample_metadata.pop("age")
+            sample_metadata["has_bulk_rna_seq"] = has_bulk_rna_seq
+            sample_metadata["has_cite_seq_data"] = has_cite_seq_data
+            sample_metadata["has_single_cell_data"] = has_single_cell_data
+            sample_metadata["has_spatial_data"] = has_spatial_data
+            sample_metadata["includes_anndata"] = include_anndata
+
+        return samples_metadata
+
+    def load_libraries_metadata(self, samples_metadata: List[Dict]):
+
+        libraries_metadata = {Sample.Modalities.SINGLE_CELL: [], Sample.Modalities.SPATIAL: []}
+
+        updated_samples_metadata = samples_metadata.copy()
+
+        for updated_sample_metadata in updated_samples_metadata:
+
+            sample_dir = self.get_sample_input_data_dir(updated_sample_metadata["scpca_sample_id"])
+            sample_cell_count_estimate = 0
+            sample_seq_units = set()
+            sample_technologies = set()
+
+            library_metadata_paths = sorted(
+                list(Path(sample_dir).glob("*_metadata.json"))
+                + list(Path(sample_dir).rglob("*_spatial/*_metadata.json"))
+            )
+            for filename_path in library_metadata_paths:
+                with open(filename_path) as library_metadata_json_file:
+                    library_json = json.load(library_metadata_json_file)
+
+                library_json["scpca_library_id"] = library_json.pop("library_id")
+                library_json["scpca_sample_id"] = library_json.pop("sample_id")
+
+                if "filtered_cells" in library_json:
+                    library_json["filtered_cell_count"] = library_json.pop("filtered_cells")
+                    sample_cell_count_estimate += library_json["filtered_cell_count"]
+
+                sample_seq_units.add(library_json["seq_unit"].strip())
+                sample_technologies.add(library_json["technology"].strip())
+
+                libraries_metadata[Sample.Modalities.SINGLE_CELL].append(
+                    library_json
+                ) if "spatial" not in str(filename_path) else libraries_metadata[
+                    Sample.Modalities.SPATIAL
+                ].append(
+                    library_json
+                )
```

Comment:
Could we rewrite this for clarity?

Reply:
Good call. I think I might have been auto-formatted here. In any case I think a normal `if else` makes more sense here then the python ternary.